api/pkg/artifact/artifact.go

```diff
@@ -2,31 +2,121 @@ package artifact
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"net/url"
 	"strings"
 
 	"cloud.google.com/go/storage"
 	"google.golang.org/api/iterator"
 )
 
+// URL contains the information needed to identify the location of an object
+// located in Google Cloud Storage.
+type URL struct {
+	// Bucket is the name of the Google Cloud Storage bucket where the object
+	// is located.
+	Bucket string
+
+	// Object is the name and or path of the object stored in the bucket. It
+	// should not start with a forward slash.
+	Object string
+}
+
 type Service interface {
+	ParseURL(gsURL string) (*URL, error)
+
+	ReadArtifact(ctx context.Context, url string) ([]byte, error)
+	WriteArtifact(ctx context.Context, url, content string) error
 	DeleteArtifact(ctx context.Context, url string) error
 }
 
 type GcsArtifactClient struct {
 	API *storage.Client
 }
 
+func NewGcsArtifactClient(api *storage.Client) Service {
+	return &GcsArtifactClient{
+		API: api,
+	}
+}
+
+// Parse parses a Google Cloud Storage string into a URL struct. The expected
+// format of the string is gs://[bucket-name]/[object-path]. If the provided
+// URL is formatted incorrectly an error will be returned.
+func (gac *GcsArtifactClient) ParseURL(gsURL string) (*URL, error) {
+	u, err := url.Parse(gsURL)
+	if err != nil {
+		return nil, err
+	}
+	if u.Scheme != "gs" {
+		return nil, err
+	}
+
+	bucket, object := u.Host, strings.TrimLeft(u.Path, "/")
+
+	if bucket == "" {
+		return nil, err
+	}
+
+	if object == "" {
+		return nil, err
+	}
+
+	return &URL{
+		Bucket: bucket,
+		Object: object,
+	}, nil
+}
+
+func (gac *GcsArtifactClient) ReadArtifact(ctx context.Context, url string) ([]byte, error) {
+	u, err := gac.ParseURL(url)
+	if err != nil {
+		return nil, err
+	}
+
+	reader, err := gac.API.Bucket(u.Bucket).Object(u.Object).NewReader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer reader.Close() //nolint:errcheck
```

Comment:
Why is the `//nolint:errcheck` added here?

Reply:
Because reader.Close() returns error and since we're not checking the value of error, linter will fail there.